src/main/java/org/jenkinsci/plugins/saml/SamlEncryptionData.java

```diff
@@ -46,11 +47,11 @@ public String getKeystorePath() {
     }
 
     public String getKeystorePassword() {
-        return keystorePassword;
+        return Secret.toString(keystorePassword);
```

Comment:
I tested it can read the old plain secrets and after that stored the encrypted password